dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1504,16 +1512,23 @@ private boolean resolveHang(long tasksCompleted, boolean queueEmpty, int poolSiz
                 Tr.event(tc, "Executor hang continued at poolSize=" + poolSize, threadPool);
             }
 
-            setPoolIncrementDecrement(poolSize);
-            if (poolSize + poolIncrement <= maxThreads && poolSize < MAX_THREADS_TO_BREAK_HANG) {
-                targetPoolSize = adjustPoolSize(poolSize, poolIncrement);
+            /**
+             * We might get here with a prior pool size change incomplete ... if so we
+             * should adjust for that
+             */
+            int hangSizeAdjustment = 0;
+            if (poolSize > targetPoolSize) {
+                setPoolIncrementDecrement(targetPoolSize);
```

Comment:
Should this be here?  We call it later as well.  Shouldn't the poolIncrement be right still from the last cycle where it was decremented?